### ISSUE TITLE
Automate testing for snippets and guides

### ISSUE THREAD

**Issue by mircubed** [dagger/dagger#3175]
The team is already working on automating testing for API endpoints, but we need automated testing for snippets in the guides as well.

**Comment by mircubed** [comment]
@aluzzardi I am assigning to you as a placeholder, but feel free to reassign and update description as you see fit.

**Comment by mircubed** [comment]
@aluzzardi @samalba should we reassign this to someone else or was this already covered in a PR?

**Comment by aluzzardi** [comment]
Partially implemented in https://github.com/dagger/dagger/pull/3634

We're building automatically them now, which ensures that at least they compile.

We don't have automated tests though, but what we have is enough short term

**Comment by shykes** [comment]
I think #4291 will get us closer to this. In theory, someone could ship other tooling improvements in parallel, but realistically I don't think that will happen given available engineering cycles. I suggest closing this in favor of #4291, wdyt @aluzzardi @marcosnils ?
